modules/miner.go

```diff
@@ -67,6 +67,12 @@ type TestMiner interface {
 	// which amounts to trying a few thousand different nonces. SolveBlock is
 	// primarily used for testing.
 	SolveBlock(types.Block, types.Target) (types.Block, bool)
+
+	// TestMiner and Miner implement the io.Closer interface with the same
+	// Close() method. Miner needs to be an io.Closer for the
+	// api.Server.Close() method; TestMiner needs to be an io.Closer for
+	// clean shutdown during testing.
+	io.Closer
```

Comment:
our style has been to write the method signature explicitly: `Close() error`. It's not a huge deal, but it signifies that calling `Close` on a module is not really the same as closing a file or a connection. As you noted in the comment, `Close` is about clean shutdown, not releasing a single resource.